packages/core/helper-plugin/src/components/Table/tests/index.test.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { ThemeProvider, lightTheme } from '@strapi/design-system';
+import { IntlProvider } from 'react-intl';
+import { Table, useTableContext } from '../index';
+
+jest.mock('../../../hooks/useQueryParams', () => jest.fn().mockReturnValue([{}, jest.fn()]));
+jest.mock('../../../features/Tracking', () => ({
+  useTracking: jest.fn(() => ({
+    trackUsage: jest.fn(),
+  })),
+}));
+
+const mockHeaders = [
+  {
+    key: '__id_key__',
+    name: 'id',
+    fieldSchema: {
+      type: 'integer',
+    },
+    metadatas: {
+      label: 'id',
+      searchable: true,
+      sortable: true,
+    },
+  },
+  {
+    key: '__short_text_key__',
+    name: 'short_text',
+    fieldSchema: {
+      type: 'string',
+    },
+    metadatas: {
+      label: 'short_text',
+      searchable: true,
+      sortable: true,
+    },
+  },
+];
+
+const mockRows = [{ id: 1 }, { id: 2 }];
+
+const BaseTestComponent = (props) => {
```

Comment:
Is this pattern used elsewhere in the codebase? I'm more used to wrappers around render()

Reply:
I'm not sure, it just made sense to me in order to avoid repeating the wrapped providers since each test uses different components.

Reply:
I would probably model it all like this:

```js
import { render as renderRTL } from '...'

const render = (props) => ({
	user: userEvent.setup(),
	...renderRTL(<Table.Root {...props} />, {
		wrapper: ({ children }) => (
			<ThemeProvider theme={lightTheme}>
				<IntlProvider locale="en" messages={{}} defaultLocale="en">
					{children}
				</IntlProvider>
			</ThemeProvider>
		)
	})
```

then in your tests you do:

```js
  it('should render with content', () => {
    render({ children: (<Table.Content contentType="Test">
          <tbody>
            <tr>
              <td>content</td>
            </tr>
          </tbody>
        </Table.Content>)
	});

    expect(screen.getByRole('cell', { name: 'content' })).toBeInTheDocument();
    expect(screen.queryByText('Loading content')).toBeNull();
  });
```

This would be more inline with what we're doing elsewhere, If you find declaring children odd you could change the API signature to

```ts
const render = (children: ReactNode, props: Table.RootProps)
```

Wdyt?

Reply:
Yeah using the wrapper makes sense and works for me. I still extracted the wrapper to its own component instead of inline so I could also use it for the renderHook in the last. Let me know what you think.